Hi @OSguard, I restarted the [travis build](https://travis-ci.org/OCA/hr/builds/30533645) after merging #10, there are quite a few pep8 errors introduced with this PR and they will need to be fixed before we can merge your code.

```
LINE_LENGTH=120

flake8 . --max-line-length=${LINE_LENGTH} --filename=__init__.py --ignore=F401 --exclude=__unported__

status1=$?

flake8 . --max-line-length=${LINE_LENGTH} --exclude=__unported__,__init__.py

./hr_public_holidays/hr_public_holidays.py:48:24: E711 comparison to None should be 'if cond is None:'
./hr_public_holidays/hr_public_holidays.py:52:47: E711 comparison to None should be 'if cond is None:'
./hr_public_holidays/hr_public_holidays.py:56:43: E128 continuation line under-indented for visual indent
./hr_public_holidays/hr_public_holidays.py:57:42: E127 continuation line over-indented for visual indent
./hr_public_holidays/hr_public_holidays.py:60:13: E128 continuation line under-indented for visual indent
./hr_public_holidays/hr_public_holidays.py:65:24: E711 comparison to None should be 'if cond is None:'
./hr_public_holidays/hr_public_holidays.py:67:45: E127 continuation line over-indented for visual indent
./hr_public_holidays/hr_public_holidays.py:68:45: E127 continuation line over-indented for visual indent
./hr_public_holidays/hr_public_holidays.py:70:45: E711 comparison to None should be 'if cond is None:'
./hr_public_holidays/hr_public_holidays.py:72:48: E127 continuation line over-indented for visual indent
./hr_public_holidays/hr_public_holidays.py:73:47: E127 continuation line over-indented for visual indent
./hr_public_holidays/hr_public_holidays.py:76:49: E127 continuation line over-indented for visual indent
./hr_public_holidays/hr_public_holidays.py:77:47: E127 continuation line over-indented for visual indent
./hr_public_holidays/hr_public_holidays.py:78:47: E128 continuation line under-indented for visual indent
./hr_public_holidays/hr_public_holidays.py:83:41: E128 continuation line under-indented for visual indent
./hr_public_holidays/hr_public_holidays.py:93:24: E711 comparison to None should be 'if cond is None:'
./hr_public_holidays/hr_public_holidays.py:97:47: E711 comparison to None should be 'if cond is None:'
./hr_public_holidays/hr_public_holidays.py:101:43: E128 continuation line under-indented for visual indent
./hr_public_holidays/hr_public_holidays.py:102:42: E127 continuation line over-indented for visual indent
./hr_public_holidays/hr_public_holidays.py:108:24: E711 comparison to None should be 'if cond is None:'
./hr_public_holidays/hr_public_holidays.py:110:44: E127 continuation line over-indented for visual indent
./hr_public_holidays/hr_public_holidays.py:111:43: E127 continuation line over-indented for visual indent
./hr_public_holidays/hr_public_holidays.py:113:45: E711 comparison to None should be 'if cond is None:'
./hr_public_holidays/hr_public_holidays.py:115:48: E127 continuation line over-indented for visual indent
./hr_public_holidays/hr_public_holidays.py:116:47: E127 continuation line over-indented for visual indent
./hr_public_holidays/hr_public_holidays.py:119:49: E127 continuation line over-indented for visual indent
./hr_public_holidays/hr_public_holidays.py:120:47: E127 continuation line over-indented for visual indent
./hr_public_holidays/hr_public_holidays.py:121:47: E128 continuation line under-indented for visual indent
./hr_public_holidays/hr_public_holidays.py:126:41: E128 continuation line under-indented for visual indent
./hr_public_holidays/hr_public_holidays.py:143:121: E501 line too long (130 > 120 characters)
```

--- a/hr_public_holidays/hr_public_holidays.py
+++ b/hr_public_holidays/hr_public_holidays.py
@@ -45,14 +45,13 @@ class hr_holidays(osv.osv):
     ]
 
     def is_public_holiday(self, cr, uid, dt, employee_id=None, context=None):
-        if employee_id == None:
-            holidays_filter = [('year', '=', dt.year), ('country_id', '=', False)]
+        employee = self.pool['hr.employee'].browse(cr, uid, employee_id)
+
+        holidays_filter = [('year', '=', dt.year)]
+        if not employee or not employee.address_id.country_id:
+            holidays_filter.append(('country_id', '=', False))
         else:
-            employee = self.pool.get('hr.employee').browse(cr, uid, employee_id)
-            if employee.address_id.country_id == None:
-                holidays_filter = [('year', '=', dt.year), ('country_id', '=', False)]
-            else:
-                holidays_filter = [('year', '=', dt.year), '|', ('country_id', '=',
+            holidays_filter += ['|', ('country_id', '=',
                                           employee.address_id.country_id.id),
                                          ('country_id', '=', False)]
 
@@ -62,23 +61,13 @@ class hr_holidays(osv.osv):
         if len(ph_ids) == 0:
             return False
 
-        if employee_id == None:
-            states_filter = [('holidays_id', 'in',
-                                            ph_ids),
-                                            ('state_ids', '=', False)]
+        states_filter = [('holidays_id', 'in', ph_ids)]
+        if not employee or not employee.address_id.state_id:
+            states_filter.append(('state_ids', '=', False))
         else:
-            if employee.address_id.state_id == None:
-                states_filter = [('holidays_id', 'in',
-                                               ph_ids),
-                                              ('state_ids', '=', False)]
-            else:
-                states_filter = [('holidays_id', 'in',
-                                                ph_ids),
-                                              '|', ('state_ids.id', '=',
-                                              employee.address_id.state_id.id),
-                                              ('state_ids', '=', False)]
+            states_filter += ['|', ('state_ids', '=', False), ('state_ids.id', '=', employee.address_id.state_id.id)]
 
-        hr_holiday_public_line_obj = self.pool.get('hr.holidays.public.line')
+        hr_holiday_public_line_obj = self.pool['hr.holidays.public.line']
         holidays_line_ids = hr_holiday_public_line_obj.search(cr, uid,
                                         states_filter)
         for line in hr_holiday_public_line_obj.browse(cr, uid, holidays_line_ids):
@@ -93,7 +82,7 @@ class hr_holidays(osv.osv):
         if employee_id == None:
             holidays_filter = [('year', '=', year), ('country_id', '=', False)]
         else:
-            employee = self.pool.get('hr.employee').browse(cr, uid, employee_id)
+            employee = self.pool['hr.employee'].browse(cr, uid, employee_id)
             if employee.address_id.country_id == None:
                 holidays_filter = [('year', '=', year), ('country_id', '=', False)]
             else:
@@ -121,7 +110,7 @@ class hr_holidays(osv.osv):
                                               employee.address_id.state_id.id),
                                               ('state_ids', '=', False)]
 
-        hr_holiday_public_line_obj = self.pool.get('hr.holidays.public.line')
+        hr_holiday_public_line_obj = self.pool['hr.holidays.public.line']
         holidays_line_ids = hr_holiday_public_line_obj.search(cr, uid,
                                         states_filter)
 

--- a/hr_public_holidays/hr_public_holidays.py
+++ b/hr_public_holidays/hr_public_holidays.py
@@ -33,7 +33,8 @@ class hr_holidays(osv.osv):
 
     _columns = {
         'year': fields.char("calendar Year", required=True),
-        'line_ids': fields.one2many('hr.holidays.public.line', 'holidays_id', 'Holiday Dates'),
+        'line_ids': fields.one2many('hr.holidays.public.line', 'holidays_id',
+                                    'Holiday Dates'),
         'country_id': fields.many2one('res.country', 'Country'),
     }
 
@@ -41,7 +42,9 @@ class hr_holidays(osv.osv):
     _order = "year"
 
     _sql_constraints = [
-        ('year_unique', 'UNIQUE(year,country_id)', _('Duplicate year and country!')),
+        ('year_unique',
+         'UNIQUE(year,country_id)',
+         _('Duplicate year and country!')),
     ]
 
     def is_public_holiday(self, cr, uid, dt, employee_id=None, context=None):
@@ -52,11 +55,11 @@ class hr_holidays(osv.osv):
             holidays_filter.append(('country_id', '=', False))
         else:
             holidays_filter += ['|', ('country_id', '=',
-                                          employee.address_id.country_id.id),
-                                         ('country_id', '=', False)]
+                                      employee.address_id.country_id.id),
+                                ('country_id', '=', False)]
 
         ph_ids = self.search(cr, uid, holidays_filter,
-            context=context)
+                             context=context)
 
         if len(ph_ids) == 0:
             return False
@@ -65,57 +68,69 @@ class hr_holidays(osv.osv):
         if not employee or not employee.address_id.state_id:
             states_filter.append(('state_ids', '=', False))
         else:
-            states_filter += ['|', ('state_ids', '=', False), ('state_ids.id', '=', employee.address_id.state_id.id)]
+            states_filter += ['|',
+                              ('state_ids', '=', False),
+                              ('state_ids.id', '=',
+                               employee.address_id.state_id.id)
+                              ]
 
         hr_holiday_public_line_obj = self.pool['hr.holidays.public.line']
-        holidays_line_ids = hr_holiday_public_line_obj.search(cr, uid,
-                                        states_filter)
-        for line in hr_holiday_public_line_obj.browse(cr, uid, holidays_line_ids):
+        holidays_line_ids = \
+            hr_holiday_public_line_obj.search(cr, uid, states_filter)
+        lines_obj = \
+            hr_holiday_public_line_obj.browse(cr, uid, holidays_line_ids)
+        for line in lines_obj:
             if date.strftime(dt, "%Y-%m-%d") == line.date:
                 return True
 
         return False
 
-    def get_holidays_list(self, cr, uid, year, employee_id=None, context=None):
+    def get_holidays_list(self, cr, uid, year, employee_id=None,
+                          context=None):
 
         res = []
-        if employee_id == None:
-            holidays_filter = [('year', '=', year), ('country_id', '=', False)]
+        if not employee_id:
+            holidays_filter = [('year', '=', year),
+                               ('country_id', '=', False)]
         else:
-            employee = self.pool['hr.employee'].browse(cr, uid, employee_id)
-            if employee.address_id.country_id == None:
-                holidays_filter = [('year', '=', year), ('country_id', '=', False)]
+            employee = \
+                self.pool['hr.employee'].browse(cr, uid, employee_id)
+            if not employee.address_id \
+                    or not employee.address_id.country_id:
+                holidays_filter = [('year', '=', year),
+                                   ('country_id', '=', False)]
             else:
-                holidays_filter = [('year', '=', year), '|', ('country_id', '=',
-                                          employee.address_id.country_id.id),
-                                         ('country_id', '=', False)]
+                holidays_filter = \
+                    [('year', '=', year), '|',
+                     ('country_id', '=', employee.address_id.country_id.id),
+                     ('country_id', '=', False)]
 
         ph_ids = self.search(cr, uid, holidays_filter, context=context)
         if len(ph_ids) == 0:
             return res
 
-        if employee_id == None:
-            states_filter = [('holidays_id', 'in',
-                                           ph_ids),
-                                          ('state_ids', '=', False)]
+        if not employee_id:
+            states_filter = [('holidays_id', 'in', ph_ids),
+                             ('state_ids', '=', False)]
         else:
-            if employee.address_id.state_id == None:
-                states_filter = [('holidays_id', 'in',
-                                               ph_ids),
-                                              ('state_ids', '=', False)]
+            if not employee.address_id or \
+                    not employee.address_id.state_id:
+                states_filter = [('holidays_id', 'in', ph_ids),
+                                 ('state_ids', '=', False)]
             else:
-                states_filter = [('holidays_id', 'in',
-                                                ph_ids),
-                                              '|', ('state_ids.id', '=',
-                                              employee.address_id.state_id.id),
-                                              ('state_ids', '=', False)]
+                states_filter = [('holidays_id', 'in', ph_ids), '|',
+                                 ('state_ids.id', '=',
+                                  employee.address_id.state_id.id),
+                                 ('state_ids', '=', False)]
 
         hr_holiday_public_line_obj = self.pool['hr.holidays.public.line']
-        holidays_line_ids = hr_holiday_public_line_obj.search(cr, uid,
-                                        states_filter)
+        holidays_line_ids = \
+            hr_holiday_public_line_obj.search(cr, uid, states_filter)
 
         [res.append(l.date)
-         for l in hr_holiday_public_line_obj.browse(cr, uid, holidays_line_ids, context=context)]
+         for l in hr_holiday_public_line_obj.browse(cr, uid,
+                                                    holidays_line_ids,
+                                                    context=context)]
         return res
 
 
@@ -127,9 +142,12 @@ class hr_holidays_line(osv.osv):
     _columns = {
         'name': fields.char('Name', size=128, required=True),
         'date': fields.date('Date', required=True),
-        'holidays_id': fields.many2one('hr.holidays.public', 'Holiday Calendar Year'),
+        'holidays_id': fields.many2one('hr.holidays.public',
+                                       'Holiday Calendar Year'),
         'variable': fields.boolean('Date may change'),
-        'state_ids': fields.many2many('res.country.state', 'hr_holiday_public_state_rel', 'line_id', 'state_id', 'Related states')
+        'state_ids': fields.many2many('res.country.state',
+                                      'hr_holiday_public_state_rel',
+                                      'line_id', 'state_id', 'Related states')
     }
 
     _order = "date, name desc"

--- a/hr_public_holidays/hr_public_holidays_view.xml
+++ b/hr_public_holidays/hr_public_holidays_view.xml
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8" ?>
 <openerp>
-    <data>
+	<data>
         
         <record id="view_public_holidays_tree" model="ir.ui.view">
             <field name="name">hr.holidays.public.tree</field>
@@ -44,15 +44,15 @@
         </record>
         
         <record id="view_public_holiday_line_form" model="ir.ui.view">
-            <field name="name">hr.holidays.public.line.form</field>
+        	<field name="name">hr.holidays.public.line.form</field>
             <field name="model">hr.holidays.public.line</field>
             <field name="arch" type="xml">
-					<form string="Public Holidays">
-                        <field name="date"/>
-	                    <field name="name"/> 
-	                    <field name="holidays_id" invisible="1"/>
-	                    <field name="state_ids" domain = "[('country_id','=',parent.country_id)]"/>  
-                	</form>
+				<form string="Public Holidays">
+                	<field name="date"/>
+	                <field name="name"/> 
+	                <field name="holidays_id" invisible="1"/>
+	                <field name="state_ids" domain = "[('country_id','=',parent.country_id)]"/>  
+            	</form>
             </field>
         </record>
         

